pgbouncer/datadog_checks/pgbouncer/pgbouncer.py

```diff
@@ -7,93 +7,55 @@
-            ('avg_query_time', ('pgbouncer.stats.avg_query_time', GAUGE)),  # >= 1.8
-        ],
-        'query': """SHOW STATS""",
-    }
-
-    POOLS_METRICS = {
-        'descriptors': [('database', 'db'), ('user', 'user')],
-        'metrics': [
-            ('cl_active', ('pgbouncer.pools.cl_active', GAUGE)),
-            ('cl_waiting', ('pgbouncer.pools.cl_waiting', GAUGE)),
-            ('sv_active', ('pgbouncer.pools.sv_active', GAUGE)),
-            ('sv_idle', ('pgbouncer.pools.sv_idle', GAUGE)),
-            ('sv_used', ('pgbouncer.pools.sv_used', GAUGE)),
-            ('sv_tested', ('pgbouncer.pools.sv_tested', GAUGE)),
-            ('sv_login', ('pgbouncer.pools.sv_login', GAUGE)),
-            ('maxwait', ('pgbouncer.pools.maxwait', GAUGE)),
-        ],
-        'query': """SHOW POOLS""",
-    }
-
-    DATABASES_METRICS = {
-        'descriptors': [('name', 'name')],
-        'metrics': [
-            ('pool_size', ('pgbouncer.databases.pool_size', GAUGE)),
-            ('max_connections', ('pgbouncer.databases.max_connections', GAUGE)),
-            ('current_connections', ('pgbouncer.databases.current_connections', GAUGE)),
-        ],
-        'query': """SHOW DATABASES""",
-    }
-
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
-        self.dbs = {}
-
-    def _get_service_checks_tags(self, host, port, database_url, tags=None):
-        if tags is None:
-            tags = []
-
-        if database_url:
-            parsed_url = urlparse(database_url)
+    def __init__(self, name, init_config, instances):
+        AgentCheck.__init__(self, name, init_config, instances)
+        self.host = self.instance.get('host', '')
+        self.port = self.instance.get('port', '')
+        self.user = self.instance.get('username', '')
+        self.password = self.instance.get('password', '')
+        self.tags = self.instance.get('tags', [])
+        self.database_url = self.instance.get('database_url')
+        self.use_cached = is_affirmative(self.instance.get('use_cached', True))
+
+        if not self.database_url:
+            if not self.host:
+                raise ConfigurationError("Please specify a PgBouncer host to connect to.")
+            if not self.user:
+                raise ConfigurationError("Please specify a user to connect to PgBouncer as.")
+        self.connection = None
+
+    def _get_service_checks_tags(self):
+        host = self.host
+        port = self.port
+        if self.database_url:
+            parsed_url = urlparse(self.database_url)
             host = parsed_url.hostname
             port = parsed_url.port
 
         service_checks_tags = ["host:%s" % host, "port:%s" % port, "db:%s" % self.DB_NAME]
-        service_checks_tags.extend(tags)
+        service_checks_tags.extend(self.tags)
         service_checks_tags = list(set(service_checks_tags))
```

Comment:
```suggestion
        service_checks_tags = service_checks_tags
```

Not sure why this is needed.

Reply:
To create a copy I guess. I'm just moving code around